broker.go

```diff
@@ -46,6 +47,33 @@ type Broker struct {
 	brokerResponseSize     metrics.Histogram
 }
 
+// SASLMechanism specifies the SASL mechanism the client uses to authenticate with the broker
+type SASLMechanism string
+
+const (
+	// SASLTypeOAuth represents the SASL/OAUTHBEARER mechanism (Kafka 2.0.0+)
+	SASLTypeOAuth = "OAUTHBEARER"
+	// SASLTypePlaintext represents the SASL/PLAIN mechanism
+	SASLTypePlaintext = "PLAIN"
+	// SASLHandshakeV0 is v0 of the Kafka SASL handshake protocol. Client and
+	// server negotiate SASL auth using opaque packets.
+	SASLHandshakeV0 = int16(0)
+	// SASLHandshakeV1 is v1 of the Kafka SASL handshake protocol. Client and
+	// server negotiate SASL by wrapping tokens with Kafka protocol headers.
+	SASLHandshakeV1 = int16(1)
+)
+
+// AccessTokenProvider is the interface that encapsulates how implementors
+// can generate access tokens for Kafka broker authentication.
+type AccessTokenProvider interface {
+	// Token returns an access token. Because this method may be called multiple
+	// times, each invocation returns a new, unexpired token. This method should
```

Comment:
Token reuse/refresh?

Reply:
My plan for this implementation is to make the user responsible for implementing the token reuse/refresh, which can be accomplished by using an OAuth client with built-in token refresh, such as [Google's OAuth2 client](https://github.com/golang/oauth2/blob/master/oauth2.go#L345).

I'm going to modify the contract to more accurately capture reuse/refresh requirements as you described.

I think it should be possible to support protocol-level token refresh by augmenting the configuration with a token expiration interval parameter that will instruct when the client should reauthenticate with the broker.